File: Armstrong Numbers/ArmstrongNumber.java
import java.util.Scanner;
 
class ArmstrongNumber
{
   public static void main(String args[])
   {
      int sum = 0, temp = 0, remainder = 0, digits = 0;
 
      Scanner sc = new Scanner(System.in);
      System.out.println("Input a number to check if it is an Armstrong number:");      
      Integer n = sc.nextInt();
 
      temp = n;
     
      // Count number of digits
      while (temp != 0) {
         digits++;
         temp = temp/10;
      }
 
      temp = n;
 
      // calculate sum of numbers to the power of digits
      while (temp != 0) {
         remainder = temp % 10;
         sum = sum + power(remainder, digits);
         temp = temp/10;
      }
      
      System.out.printf("%d %s an Armstrong number.\n", n, (sum == n)? "is" : "isn't");  
   }
   
   static int power(int n, int r) {
      int c, p = 1;
 
      for (c = 1; c <= r; c++)
         p = p * n;
 
      return p;  
   }
}

File: Armstrong Numbers/ArmstrongNumber.kt
/**
 * @created October 2, 2019
 * Armstrong Number logic in Kotlin
 * @explanation Armstrong number is a number that is equal to the sum of cubes of its digits.
 */

class ArmstrongNumber {

    fun solve() {
        var sum = 0
        var remainder: Int

        println("Input a number to check if it is an Armstrong number")
        val n = readLine()!!.toInt()

        var temporal = n
        while (temporal != 0) {
            remainder = temporal % 10
            sum += (remainder * remainder * remainder)
            temporal /= 10
        }

        if (n == sum) {
            println("$n is an Armstrong number.")
        } else {
            println("$n isn't an Armstrong number.")
        }
    }

}

fun main() {
    ArmstrongNumber().solve()
}